superset-frontend/src/explore/components/controls/CollectionControl/CollectionControl.test.tsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import CollectionControl from '.';
+
+jest.mock('@superset-ui/chart-controls', () => ({
+  InfoTooltipWithTrigger: (props: any) => (
+    <button
+      onClick={props.onClick}
+      type="button"
+      data-icon={props.icon}
+      data-tooltip={props.tooltip}
+    >
+      {props.label}
+    </button>
+  ),
+}));
+
+jest.mock('..', () => ({
+  __esModule: true,
+  default: {
+    TestControl: (props: any) => (
+      <button
+        type="button"
+        onClick={() => props.onChange(0, 'update')}
+        data-test="TestControl"
```

Comment:
Small nit. I think there is a convention of naming the `data-test` with hyphen separated names or just one lowercase word

Reply:
In this case it is a mock. I put the data-test equal to the name of the component that is being replaced